sphinx/builders/html/_assets.py

```diff
@@ -136,11 +147,40 @@ def _file_checksum(outdir: Path, filename: str | os.PathLike[str]) -> str:
     if '?' in filename:
         msg = f'Local asset file paths must not contain query strings: {filename!r}'
         raise ThemeError(msg)
+    algorithms = {
+        'sha256': sha256,
+    }
+    checksums = []
     try:
-        # Remove all carriage returns to avoid checksum differences
-        content = outdir.joinpath(filename).read_bytes().translate(None, b'\r')
```

Comment:
@AA-Turner although I closed this pull request, there's one detail it uncovered that seems worth mentioning: our test suite fails _on Windows only_ without the removal of these carriage-returns, but the reason surprised me: it's due to `git` checking out different styles of newline (and therefore different content as input to the checksum algorithm) -- `CRLF` on Windows, and `LF` on Linux -- by default.

It's possible to configure `git` to use specific line-endings per-file on checkout - I used a directory-specific `.gitattributes` file to achieve that, and configuring at least one example of each line format:

https://github.com/sphinx-doc/sphinx/pull/12231/commits/d123fb4c806a58c3bc7bbeb60f485e914f94d953

It might be worth extracting and applying that separately so that we have specific test coverage for it; let me know what you think when you have time.